flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/AbstractRocksDBState.java

```diff
@@ -190,4 +197,12 @@ protected void writeKeyWithGroupAndNamespace(
 		RocksDBKeySerializationUtils.writeKey(key, keySerializer, keySerializationStream, keySerializationDataOutputView, ambiguousKeyPossible);
 		RocksDBKeySerializationUtils.writeNameSpace(namespace, namespaceSerializer, keySerializationStream, keySerializationDataOutputView, ambiguousKeyPossible);
 	}
+
+	protected V getDefaultValue() {
```

Comment:
this method is duplicated among some impl classes. We can move it to `InternalKvState` as a [default method](https://docs.oracle.com/javase/tutorial/java/IandI/defaultmethods.html).

Reply:
Not too sure about this one.

That would require introducing 2 methods in the `InternalKvState`:
1. A getter method that returns the default value.
2. A default method that actually does the serialization copying of the default value (the current `getDefaultValue` method).

Reply:
Since this PR is blocking a bug for the 1.5 release, I'll proceed to merge this as it is.
@bowenli86 Perhaps we can open a separate JIRA for this to keep this in mind?